src/toil/batchSystems/parasol.py

```diff
@@ -132,23 +147,31 @@ def issueBatchJob(self, command, memory, cores, disk):
         """Issues parasol with job commands.
         """
         self.checkResourceRequest(memory, cores, disk)
+        #look for a batch for jobs with these resource requirements
+        batch = None
+        for i in range(len(self.batches)):
+            if self.batches[i].cores == cores and self.batches[i].memory == memory:
+                batch = self.batches[i]
+        if batch is None:
+            batch = ParasolBatch(batchSystem = self, cores = cores, memory = memory)
+            self.batches.append(batch)
         pattern = re.compile("your job ([0-9]+).*")
-        parasolCommand = "%s -verbose -ram=%i -cpu=%i -results=%s add job '%s'" % (self.parasolCommand, memory, cores, self.parasolResultsFile, command)
+        parasolCommand = "%s -verbose -ram=%i -cpu=%i -results=%s add job '%s'" % (self.parasolCommand, memory, cores, batch.resultsFile, command)
 
         #Deal with the cpus
         self.usedCpus += cores
         while True: #Process finished results with no wait
             try:
-               jobID = self.outputQueue1.get_nowait()
+               jobID = batch.outputQueue1.get_nowait()
```

Comment:
Let's make this line the only line in the try: block and move the remaining three lines into an else: clause. That's safer and easier to read.